pkg/api/namespace/formatter.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/harvester/harvester/pkg/util"
+)
+
+const (
+	updateResourceQuotaAction = "updateResourceQuota"
+	deleteResourceQuotaAction = "deleteResourceQuota"
+)
+
+type Handler struct {
+	resourceQuotaClient ctlharvesterv1.ResourceQuotaClient
+	clientSet           kubernetes.Clientset
+
+	ctx context.Context
+}
+
+type nsformatter struct {
+	clientSet kubernetes.Clientset
+}
+
+func (nf *nsformatter) formatter(request *types.APIRequest, resource *types.RawResource) {
+	resource.Actions = make(map[string]string, 1)
+	if request.AccessControl.CanUpdate(request, resource.APIObject, resource.Schema) != nil {
+		return
+	}
+
+	if ok, err := apiutil.CanUpdateResourceQuota(nf.clientSet, request.Namespace, request.GetUser()); err != nil {
+		logrus.WithError(err).WithFields(logrus.Fields{
+			"namespace": request.Namespace,
+			"user":      request.GetUser(),
+		}).Error("Failed to check update resource quota")
+	} else if ok {
+		resource.AddAction(request, updateResourceQuotaAction)
+		resource.AddAction(request, deleteResourceQuotaAction)
+	}
+}
+
+func (h *Handler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	if err := h.do(rw, req); err != nil {
+		status := http.StatusInternalServerError
+		if e, ok := err.(*apierror.APIError); ok {
+			status = e.Code.Status
+		}
+		rw.WriteHeader(status)
+		_, _ = rw.Write([]byte(err.Error()))
+		return
+	}
+	rw.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) do(_ http.ResponseWriter, req *http.Request) error {
+	vars := util.EncodeVars(mux.Vars(req))
+	action := vars["action"]
+	namespace := vars["name"] // since this is namespace api, so name is namespace here
+
+	user, ok := request.UserFrom(req.Context())
+	if !ok {
+		return apierror.NewAPIError(validation.Unauthorized, "failed to get user from request")
+	}
+
+	switch action {
+	case updateResourceQuotaAction:
+		if ok, err := apiutil.CanUpdateResourceQuota(h.clientSet, namespace, user.GetName()); err != nil {
+			return apierror.NewAPIError(validation.ServerError, fmt.Sprintf("Failed to check permission: %v", err))
+		} else if !ok {
+			return apierror.NewAPIError(validation.PermissionDenied, "User does not have permission to update resource quota")
+		}
+
+		var updateResourceQuotaInput UpdateResourceQuotaInput
+		if err := json.NewDecoder(req.Body).Decode(&updateResourceQuotaInput); err != nil {
+			return apierror.NewAPIError(validation.InvalidBodyContent, fmt.Sprintf("Failed to decode request body: %v ", err))
+		}
+		return h.updateResourceQuota(namespace, updateResourceQuotaInput)
+	case deleteResourceQuotaAction:
+		if ok, err := apiutil.CanUpdateResourceQuota(h.clientSet, namespace, user.GetName()); err != nil {
+			return apierror.NewAPIError(validation.ServerError, fmt.Sprintf("Failed to check permission: %v", err))
+		} else if !ok {
+			return apierror.NewAPIError(validation.PermissionDenied, "User does not have permission to update resource quota")
+		}
```

Comment:
I would prefer to keep it, because these API will update `resourcequotas.harvesterhci.io`. The default authentication for VM/Namespace (by steve server) can't work for `resourcequotas.harvesterhci.io`.

Only changing `resourcequotas.harvesterhci.io` APIs need this. Other changing VM/Namespace APIs don't need it.